buildbot/check-clang.sh

```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
+
+PROJECT=
+BRANCH=
+BUILD_NUMBER=
+PR_NUMBER=
+
+# $1 exit code
+# $2 error message
+exit_if_err()
+{
+    if [ $1 -ne 0  ]; then
+        echo "Error: $2"
+        exit $1
+    fi
+}
+
+unset OPTIND
+while getopts ":p:b:r:n:" option; do
+    case $option in
+        p) PROJECT=$OPTARG ;;
+        b) BRANCH=$OPTARG ;;
+        n) BUILD_NUMBER=$OPTARG ;;
+        r) PR_NUMBER=$OPTARG ;;
+    esac
+done && shift $(($OPTIND - 1))
+
+# we're in llvm.obj dir
+BUILD_DIR=${PWD}
+
+make check-clang VERBOSE=1 LIT_ARGS="-v -j `nproc`"
```

Comment:
Is it possible to make this utility configurable?
I use ninja to build LLVM and run the tests and I'd like to use these scripts on my local machine to reproduce the results/debug/etc.
I think you can re-use cmake for this: `cmake --build <> --taget <>`

Reply:
I think the scripts are for builders on Buildbot, Buildbot will run these scripts during build. As per @vladimirlaz , developer will control these scripts, you can change to ninja as you wish, but you need to change all of the scripts, e.g. conigure.sh, compile.sh

Reply:
I think the scripts are for builders on Buildbot, Buildbot will run these scripts during build. As per @vladimirlaz , developer will control these scripts, you can change to ninja as you wish, but you need to change all of the scripts, e.g. conigure.sh, compile.sh

Reply:
Do we need these complication? If you want to reproduce failure it is better to have exact the same configuration including tooling.
Locally developer can use any tools chain he needs.